lmfit/minimizer.py

```diff
@@ -282,11 +290,16 @@ def __residual(self, fvars):
         if self._abort:
             return None
         params = self.result.params
-        for name, val in zip(self.result.var_names, fvars):
-            params[name].value = params[name].from_internal(val)
+
+        if apply_scaling:
+            for name, val in zip(self.result.var_names, fvars):
+                params[name].value = params[name].from_internal(val)
+                params.update_constraints()
+        else:
+            for name, val in zip(self.result.var_names, fvars):
+                params[name].value = val
         self.result.nfev += 1
 
-        params.update_constraints()
```

Comment:
Given my comments above this line needs to be added back in - `update_constraints` needs to be called even if `apply_scaling is False`.